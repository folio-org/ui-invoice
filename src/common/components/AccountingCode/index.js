export { default as AccountingCodeValue } from './AccountingCodeValue';
export { default as FieldAccountingCode } from './FieldAccountingCode';
