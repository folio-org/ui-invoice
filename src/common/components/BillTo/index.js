export { default as BillToValue } from './BillToValue';
