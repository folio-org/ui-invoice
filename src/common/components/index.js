export { default as ApprovedBy } from './ApprovedBy';
export {
  CredentialsField,
  CredentialsFieldsGroup,
  CredentialsToggle,
} from './credentials';
export { default as DuplicateInvoiceList } from './DuplicateInvoiceList';
export * from './FieldFiscalYear';
export * from './FiscalYearValue';
export * from './VendorPrimaryAddress';
export * from './Status';
