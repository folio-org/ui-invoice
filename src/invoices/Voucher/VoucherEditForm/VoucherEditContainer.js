import React, { useCallback, useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import ReactRouterPropTypes from 'react-router-prop-types';

import {
  LoadingView,
} from '@folio/stripes/components';
import {
  stripesConnect,
} from '@folio/stripes/core';
import {
  getConfigSetting,
  useShowCallout,
} from '@folio/stripes-acq-components';

import {
  invoiceResource,
  VOUCHER_BY_ID,
  configVoucherNumber,
} from '../../../common/resources';
import VoucherEditForm from './VoucherEditForm';

const VoucherEditContainer = ({
  match: { params },
  mutator,
  history,
  location,
}) => {
  const [voucher, setVoucher] = useState();
  const [vendorInvoiceNo, setVoucherInvoiceNo] = useState();
  const [isAllowVoucherNumberEdit, setAllowVoucherNumberEdit] = useState();
  const showCallout = useShowCallout();

  useEffect(() => {
    const voucherPromise = mutator.voucher.GET();
    const invoicePromise = mutator.invoice.GET();
    const configVoucherPromise = mutator.configVoucherNumber.GET();

    Promise.all([invoicePromise, configVoucherPromise, voucherPromise])
      .then(([invoiceResponse, configVoucherResponse, voucherResponse]) => {
        const { allowVoucherNumberEdit } = getConfigSetting(configVoucherResponse);

        setVoucherInvoiceNo(invoiceResponse.vendorInvoiceNo);
        setAllowVoucherNumberEdit(allowVoucherNumberEdit);
        setVoucher(voucherResponse);
      })
      .catch(() => {
        showCallout({ messageId: 'ui-invoice.errors.voucherLoadingFailed', type: 'error' });
      });
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const closeVoucherForm = useCallback(
    () => {
      const _path = `/invoice/view/${params.id}/voucher/${params.voucherId}/view`;

      history.push({
        pathname: _path,
        search: location.search,
      });
    },
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [params.id, location.search],
  );

  const saveVoucher = useCallback(
    (vouch) => {
      mutator.voucher.PUT(vouch)
        .then(() => {
          showCallout({ messageId: 'ui-invoice.invoice.details.voucher.save.success' });
          closeVoucherForm();
        })
        .catch(() => {
          showCallout({ messageId: 'ui-invoice.errors.voucherHasNotBeenSaved', type: 'error' });
        });
    },
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [closeVoucherForm, showCallout],
  );

  if (!(voucher && vendorInvoiceNo)) {
    return (
      <LoadingView onClose={closeVoucherForm} />
    );
  }

  return (
    <VoucherEditForm
      initialValues={voucher}
      onSubmit={saveVoucher}
      onCancel={closeVoucherForm}
      vendorInvoiceNo={vendorInvoiceNo}
      isAllowVoucherNumberEdit={isAllowVoucherNumberEdit}
    />
  );
};

VoucherEditContainer.manifest = Object.freeze({
  voucher: {
    ...VOUCHER_BY_ID,
    accumulate: true,
  },
  invoice: {
    ...invoiceResource,
    accumulate: true,
  },
  configVoucherNumber: {
    ...configVoucherNumber,
    accumulate: true,
  },
});

VoucherEditContainer.propTypes = {
  match: ReactRouterPropTypes.match,
  mutator: PropTypes.object.isRequired,
  history: ReactRouterPropTypes.history,
  location: ReactRouterPropTypes.location,
};

export default stripesConnect(VoucherEditContainer);
