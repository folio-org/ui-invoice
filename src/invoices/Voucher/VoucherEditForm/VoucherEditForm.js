import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { Field } from 'react-final-form';

import stripesFinalForm from '@folio/stripes/final-form';
import {
  Row,
  Col,
  TextField,
  Pane,
  Paneset,
  KeyValue,
} from '@folio/stripes/components';
import {
  FieldDatepickerFinal,
  FormFooter,
} from '@folio/stripes-acq-components';
import {
  AppIcon,
} from '@folio/stripes/core';

const VoucherEditForm = ({
  handleSubmit,
  submitting,
  pristine,
  onCancel,
  initialValues,
  vendorInvoiceNo,
  isAllowVoucherNumberEdit,
}) => {
  const paneFooter = (
    <FormFooter
      id="save-voucher-button"
      label={<FormattedMessage id="ui-invoice.saveAndClose" />}
      pristine={pristine}
      submitting={submitting}
      handleSubmit={handleSubmit}
      onCancel={onCancel}
    />
  );

  return (
    <Paneset
      data-test-edit-voucher-form
    >
      <Pane
        appIcon={<AppIcon app="invoice" size="small" />}
        defaultWidth="fill"
        dismissible
        id="pane-edit-voucher"
        footer={paneFooter}
        onClose={onCancel}
        paneSub={
          <FormattedMessage
            id="ui-invoice.voucher.paneSubTitle"
            values={{ voucherNumber: initialValues.voucherNumber }}
          />
        }
        paneTitle={
          <FormattedMessage
            id="ui-invoice.voucher.paneTitle"
            values={{ vendorInvoiceNo }}
          />
        }
      >
        <Row>
          <Col xs={3}>
            {isAllowVoucherNumberEdit
              ? (
                <Field
                  component={TextField}
                  id="voucherNumber"
                  label={<FormattedMessage id="ui-invoice.invoice.details.voucher.voucherNumber" />}
                  name="voucherNumber"
                />
              )
              : (
                <KeyValue
                  label={<FormattedMessage id="ui-invoice.invoice.details.voucher.voucherNumber" />}
                  value={initialValues.voucherNumber}
                />
              )
            }
          </Col>
          <Col xs={3}>
            <Field
              component={TextField}
              id="disbursementNumber"
              label={<FormattedMessage id="ui-invoice.invoice.details.voucher.disbursementNumber" />}
              name="disbursementNumber"
            />
          </Col>
          <Col xs={3}>
            <FieldDatepickerFinal
              label={<FormattedMessage id="ui-invoice.invoice.details.voucher.disbursementDate" />}
              name="disbursementDate"
            />
          </Col>
          <Col xs={3}>
            <Field
              component={TextField}
              id="disbursementAmount"
              label={<FormattedMessage id="ui-invoice.invoice.details.voucher.disbursementAmount" />}
              name="disbursementAmount"
            />
          </Col>
        </Row>
      </Pane>
    </Paneset>
  );
};

VoucherEditForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
  submitting: PropTypes.bool.isRequired,
  pristine: PropTypes.bool.isRequired,
  isAllowVoucherNumberEdit: PropTypes.bool.isRequired,
  initialValues: PropTypes.object.isRequired,
  vendorInvoiceNo: PropTypes.string.isRequired,
};

export default stripesFinalForm({
  navigationCheck: true,
})(VoucherEditForm);
