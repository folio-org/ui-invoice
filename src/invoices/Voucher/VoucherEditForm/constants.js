export const EDIT_VOUCHER_FORM = 'editVoucher';
