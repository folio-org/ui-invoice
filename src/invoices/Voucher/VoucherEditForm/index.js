export { default as VoucherEditLayer } from './VoucherEditLayer';
