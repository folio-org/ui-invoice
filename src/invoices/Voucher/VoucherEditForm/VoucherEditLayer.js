import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import ReactRouterPropTypes from 'react-router-prop-types';
import { get } from 'lodash';

import { Layer } from '@folio/stripes/components';
import {
  stripesConnect,
} from '@folio/stripes/core';

import { LoadingPane } from '../../../common/components';
import {
  invoiceResource,
  VOUCHER_BY_ID,
  configVoucherNumber,
} from '../../../common/resources';
import { getIsAllowVoucherNumberEdit } from '../utils';
import VoucherEditForm from './VoucherEditForm';

const VoucherEditLayer = ({ match: { params }, mutator, resources }) => {
  const voucher = get(resources, ['voucher', 'records', 0]);
  const vendorInvoiceNo = get(resources, ['invoice', 'records', 0, 'vendorInvoiceNo']);
  const isAllowVoucherNumberEdit = getIsAllowVoucherNumberEdit(get(resources, ['configVoucherNumber', 'records', 0]));
  const isLoading = !get(resources, ['voucher', 'hasLoaded']);
  const closeVoucherForm = useCallback(
    () => {
      const _path = `/invoice/view/${params.id}/voucher/${params.voucherId}/view`;

      mutator.query.update({ _path });
    },
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [params.id],
  );

  const saveVoucher = useCallback(
    (vouch) => {
      mutator.voucher.PUT(vouch).then(closeVoucherForm);
    },
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [mutator.voucher],
  );

  if (isLoading) {
    return (
      <Layer isOpen>
        <LoadingPane onClose={closeVoucherForm} />
      </Layer>
    );
  }

  return (
    <Layer isOpen>
      <VoucherEditForm
        initialValues={voucher}
        onSubmit={saveVoucher}
        onCancel={closeVoucherForm}
        vendorInvoiceNo={vendorInvoiceNo}
        isAllowVoucherNumberEdit={isAllowVoucherNumberEdit}
      />
    </Layer>
  );
};

VoucherEditLayer.manifest = Object.freeze({
  voucher: VOUCHER_BY_ID,
  invoice: invoiceResource,
  configVoucherNumber,
  query: {},
});

VoucherEditLayer.propTypes = {
  match: ReactRouterPropTypes.match,
  mutator: PropTypes.object.isRequired,
  resources: PropTypes.object.isRequired,
};

export default stripesConnect(VoucherEditLayer);
