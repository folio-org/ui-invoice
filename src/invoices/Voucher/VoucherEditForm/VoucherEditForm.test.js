import { MemoryRouter } from 'react-router-dom';
import { useHistory } from 'react-router';

import user from '@folio/jest-config-stripes/testing-library/user-event';
import { render, screen } from '@folio/jest-config-stripes/testing-library/react';
import { HasCommand } from '@folio/stripes/components';

import { batchVoucher } from '../../../../test/jest/fixtures';

import VoucherEditForm from './VoucherEditForm';

jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useHistory: jest.fn(),
}));
jest.mock('@folio/stripes-components/lib/Commander', () => ({
  HasCommand: jest.fn(({ children }) => <div>{children}</div>),
}));

const defaultProps = {
  onSubmit: jest.fn(),
  initialValues: batchVoucher,
  onCancel: jest.fn(),
  vendorInvoiceNo: 'vendorInvoiceNo',
};
const renderVoucherEditForm = (props = defaultProps) => render(
  <VoucherEditForm {...props} />,
  { wrapper: MemoryRouter },
);

describe('VoucherEditForm', () => {
  it('should render correct voucher form structure', () => {
    const { asFragment } = renderVoucherEditForm();

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correct voucher form structure with editable voucher number', () => {
    const { asFragment } = renderVoucherEditForm({
      ...defaultProps,
      isAllowVoucherNumberEdit: true,
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it('should call onCancel prop when cancel button is pressed', async () => {
    const onCancel = jest.fn();

    renderVoucherEditForm({
      ...defaultProps,
      onCancel,
    });

    await user.click(screen.getByText('stripes-acq-components.FormFooter.cancel'));

    expect(onCancel).toHaveBeenCalled();
  });

  describe('Shortcuts', () => {
    beforeEach(() => {
      HasCommand.mockClear();
    });

    it('should cancel form when cancel shortcut is called', () => {
      defaultProps.onCancel.mockClear();

      renderVoucherEditForm();

      HasCommand.mock.calls[0][0].commands.find(c => c.name === 'cancel').handler();

      expect(defaultProps.onCancel).toHaveBeenCalled();
    });

    it('should navigate to list view when search shortcut is called', () => {
      const pushMock = jest.fn();

      useHistory.mockClear().mockReturnValue({
        push: pushMock,
      });

      renderVoucherEditForm();
      HasCommand.mock.calls[0][0].commands.find(c => c.name === 'search').handler();

      expect(pushMock).toHaveBeenCalledWith('/invoice');
    });
  });
});
