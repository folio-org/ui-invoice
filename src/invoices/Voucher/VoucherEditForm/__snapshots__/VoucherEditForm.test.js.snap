// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoucherEditForm should render correct voucher form structure 1`] = `
<DocumentFragment>
  <div>
    <div
      class="paneset"
      data-test-edit-voucher-form="true"
    >
      <section
        class="pane focusIndicator"
        id="pane-edit-voucher"
      >
        <div
          aria-labelledby="paneHeaderpane-edit-voucher-pane-title"
          class="paneHeader"
          data-test-pane-header="true"
          id="paneHeaderpane-edit-voucher"
          role="region"
          tabindex="-1"
        >
          <div
            class="paneHeaderButtonsArea first"
          >
            <div
              class="paneMenu"
            >
              <button
                aria-label="stripes-components.closeItem"
                class="iconButton medium paneHeaderCloseIcon PaneHeaderIconButton paneHeaderIconButton"
                data-test-pane-header-dismiss-button="true"
                icon="times"
                type="button"
              >
                <span
                  class="iconButtonInner mediumOffset PaneHeaderIconButton__inner"
                  data-test-pane-header-dismiss-button="true"
                >
                  <span>
                    Icon
                  </span>
                </span>
              </button>
            </div>
          </div>
          <div
            class="paneHeaderCenter"
          >
            <div
              class="paneHeaderCenterInner"
              id="paneHeaderpane-edit-voucher-pane-title"
            >
              <h2
                class="paneTitle"
                data-test-pane-header-title="true"
              >
                <div
                  class="appIcon small icon-alignment-center"
                >
                  <span
                    aria-hidden="true"
                    class="icon"
                  />
                </div>
                <span
                  class="paneTitleLabel"
                >
                  ui-invoice.voucher.paneTitle
                </span>
              </h2>
              <p
                class="paneSub"
                data-test-pane-header-sub="true"
                id="paneHeaderpane-edit-voucher-subtitle"
              >
                <span>
                  ui-invoice.voucher.paneSubTitle
                </span>
              </p>
            </div>
          </div>
        </div>
        <div
          class="paneContent hasPadding"
          id="pane-edit-voucher-content"
          tabindex="-1"
        >
          <div
            class="row"
          >
            <div
              class="col-xs-3"
            >
              <div
                class="kvRoot"
              >
                <div
                  class="kvLabel"
                >
                  ui-invoice.invoice.details.voucher.voucherNumber
                </div>
                <div
                  class="kvValue"
                  data-test-kv-value="true"
                >
                  1023
                </div>
              </div>
            </div>
            <div
              class="col-xs-3"
            >
              <div
                class="textField"
              >
                <label
                  class="label"
                  for="disbursementNumber"
                >
                  ui-invoice.invoice.details.voucher.disbursementNumber
                </label>
                <div
                  class="formControl inputGroup"
                >
                  <input
                    aria-invalid="false"
                    autocapitalize="off"
                    autocomplete="off"
                    autocorrect="off"
                    error=""
                    id="disbursementNumber"
                    name="disbursementNumber"
                    spellcheck="false"
                    type="text"
                    value="EFT546789"
                    warning=""
                  />
                </div>
                <div
                  role="alert"
                />
              </div>
            </div>
            <div
              class="col-xs-3"
            >
              <div
                class="container"
              >
                <div
                  class="container"
                  data-test-datepicker-container="true"
                >
                  <div
                    class="textField"
                  >
                    <label
                      class="label"
                      for="dp-2"
                    >
                      ui-invoice.invoice.details.voucher.disbursementDate
                    </label>
                    <div
                      class="formControl inputGroup"
                    >
                      <input
                        aria-invalid="false"
                        autocapitalize="off"
                        autocomplete="off"
                        autocorrect="off"
                        id="dp-2"
                        name="disbursementDate"
                        placeholder="MM/DD/YYYY"
                        spellcheck="false"
                        type="text"
                        value="12/06/2019"
                      />
                      <div
                        class="endControls"
                      >
                        <div
                          class="controlGroup"
                        >
                          <button
                            aria-label="stripes-components.clearFieldValue"
                            class="iconButton medium"
                            data-test-clear="true"
                            icon="times-circle-solid"
                            id="datepicker-clear-button-dp-2"
                            tabindex="-1"
                            type="button"
                          >
                            <span
                              class="iconButtonInner mediumOffset"
                              data-test-clear="true"
                            >
                              <span>
                                Icon
                              </span>
                            </span>
                          </button>
                          <button
                            aria-haspopup="true"
                            aria-label="stripes-components.showOrHideDatepicker"
                            class="iconButton medium"
                            data-test-calendar-button="true"
                            icon="calendar"
                            id="datepicker-toggle-calendar-button-dp-2"
                            type="button"
                          >
                            <span
                              class="iconButtonInner mediumOffset"
                              data-test-calendar-button="true"
                            >
                              <span>
                                Icon
                              </span>
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                    <div
                      role="alert"
                    />
                  </div>
                  <input
                    data-test-datepicker-hidden-input="true"
                    hidden=""
                    type="text"
                    value="2019-12-06"
                  />
                </div>
              </div>
            </div>
            <div
              class="col-xs-3"
            >
              <div
                class="textField"
              >
                <label
                  class="label"
                  for="disbursementAmount"
                >
                  ui-invoice.invoice.details.voucher.disbursementAmount
                </label>
                <div
                  class="formControl inputGroup"
                >
                  <input
                    aria-invalid="false"
                    autocapitalize="off"
                    autocomplete="off"
                    autocorrect="off"
                    error=""
                    id="disbursementAmount"
                    name="disbursementAmount"
                    spellcheck="false"
                    type="text"
                    value="23.45"
                    warning=""
                  />
                </div>
                <div
                  role="alert"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="paneFooter"
        >
          <div
            class="paneFooterContent"
          >
            <span
              class="renderStart"
              data-test-pane-footer-start="true"
            >
              <button
                class="button default mega"
                data-test-cancel-button="true"
                type="button"
              >
                <span
                  class="inner"
                >
                  stripes-acq-components.FormFooter.cancel
                </span>
              </button>
            </span>
            <span
              data-test-pane-footer-end="true"
            >
              <button
                class="button primary mega"
                data-test-save-button="true"
                disabled=""
                id="save-voucher-button"
                type="submit"
              >
                <span
                  class="inner"
                >
                  stripes-components.saveAndClose
                </span>
              </button>
            </span>
          </div>
        </div>
      </section>
    </div>
    <div
      class="container"
    >
      <div
        class="cursorHover"
        hidden=""
        style="left: -10px;"
      >
        <div
          class="cursor"
        >
          Â 
        </div>
      </div>
    </div>
  </div>
  <div />
</DocumentFragment>
`;

exports[`VoucherEditForm should render correct voucher form structure with editable voucher number 1`] = `
<DocumentFragment>
  <div>
    <div
      class="paneset"
      data-test-edit-voucher-form="true"
    >
      <section
        class="pane focusIndicator"
        id="pane-edit-voucher"
      >
        <div
          aria-labelledby="paneHeaderpane-edit-voucher-pane-title"
          class="paneHeader"
          data-test-pane-header="true"
          id="paneHeaderpane-edit-voucher"
          role="region"
          tabindex="-1"
        >
          <div
            class="paneHeaderButtonsArea first"
          >
            <div
              class="paneMenu"
            >
              <button
                aria-label="stripes-components.closeItem"
                class="iconButton medium paneHeaderCloseIcon PaneHeaderIconButton paneHeaderIconButton"
                data-test-pane-header-dismiss-button="true"
                icon="times"
                type="button"
              >
                <span
                  class="iconButtonInner mediumOffset PaneHeaderIconButton__inner"
                  data-test-pane-header-dismiss-button="true"
                >
                  <span>
                    Icon
                  </span>
                </span>
              </button>
            </div>
          </div>
          <div
            class="paneHeaderCenter"
          >
            <div
              class="paneHeaderCenterInner"
              id="paneHeaderpane-edit-voucher-pane-title"
            >
              <h2
                class="paneTitle"
                data-test-pane-header-title="true"
              >
                <div
                  class="appIcon small icon-alignment-center"
                >
                  <span
                    aria-hidden="true"
                    class="icon"
                  />
                </div>
                <span
                  class="paneTitleLabel"
                >
                  ui-invoice.voucher.paneTitle
                </span>
              </h2>
              <p
                class="paneSub"
                data-test-pane-header-sub="true"
                id="paneHeaderpane-edit-voucher-subtitle"
              >
                <span>
                  ui-invoice.voucher.paneSubTitle
                </span>
              </p>
            </div>
          </div>
        </div>
        <div
          class="paneContent hasPadding"
          id="pane-edit-voucher-content"
          tabindex="-1"
        >
          <div
            class="row"
          >
            <div
              class="col-xs-3"
            >
              <div
                class="textField"
              >
                <label
                  class="label"
                  for="voucherNumber"
                >
                  ui-invoice.invoice.details.voucher.voucherNumber
                </label>
                <div
                  class="formControl inputGroup"
                >
                  <input
                    aria-invalid="false"
                    autocapitalize="off"
                    autocomplete="off"
                    autocorrect="off"
                    error=""
                    id="voucherNumber"
                    name="voucherNumber"
                    spellcheck="false"
                    type="text"
                    value="1023"
                    warning=""
                  />
                </div>
                <div
                  role="alert"
                />
              </div>
            </div>
            <div
              class="col-xs-3"
            >
              <div
                class="textField"
              >
                <label
                  class="label"
                  for="disbursementNumber"
                >
                  ui-invoice.invoice.details.voucher.disbursementNumber
                </label>
                <div
                  class="formControl inputGroup"
                >
                  <input
                    aria-invalid="false"
                    autocapitalize="off"
                    autocomplete="off"
                    autocorrect="off"
                    error=""
                    id="disbursementNumber"
                    name="disbursementNumber"
                    spellcheck="false"
                    type="text"
                    value="EFT546789"
                    warning=""
                  />
                </div>
                <div
                  role="alert"
                />
              </div>
            </div>
            <div
              class="col-xs-3"
            >
              <div
                class="container"
              >
                <div
                  class="container"
                  data-test-datepicker-container="true"
                >
                  <div
                    class="textField"
                  >
                    <label
                      class="label"
                      for="dp-5"
                    >
                      ui-invoice.invoice.details.voucher.disbursementDate
                    </label>
                    <div
                      class="formControl inputGroup"
                    >
                      <input
                        aria-invalid="false"
                        autocapitalize="off"
                        autocomplete="off"
                        autocorrect="off"
                        id="dp-5"
                        name="disbursementDate"
                        placeholder="MM/DD/YYYY"
                        spellcheck="false"
                        type="text"
                        value="12/06/2019"
                      />
                      <div
                        class="endControls"
                      >
                        <div
                          class="controlGroup"
                        >
                          <button
                            aria-label="stripes-components.clearFieldValue"
                            class="iconButton medium"
                            data-test-clear="true"
                            icon="times-circle-solid"
                            id="datepicker-clear-button-dp-5"
                            tabindex="-1"
                            type="button"
                          >
                            <span
                              class="iconButtonInner mediumOffset"
                              data-test-clear="true"
                            >
                              <span>
                                Icon
                              </span>
                            </span>
                          </button>
                          <button
                            aria-haspopup="true"
                            aria-label="stripes-components.showOrHideDatepicker"
                            class="iconButton medium"
                            data-test-calendar-button="true"
                            icon="calendar"
                            id="datepicker-toggle-calendar-button-dp-5"
                            type="button"
                          >
                            <span
                              class="iconButtonInner mediumOffset"
                              data-test-calendar-button="true"
                            >
                              <span>
                                Icon
                              </span>
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                    <div
                      role="alert"
                    />
                  </div>
                  <input
                    data-test-datepicker-hidden-input="true"
                    hidden=""
                    type="text"
                    value="2019-12-06"
                  />
                </div>
              </div>
            </div>
            <div
              class="col-xs-3"
            >
              <div
                class="textField"
              >
                <label
                  class="label"
                  for="disbursementAmount"
                >
                  ui-invoice.invoice.details.voucher.disbursementAmount
                </label>
                <div
                  class="formControl inputGroup"
                >
                  <input
                    aria-invalid="false"
                    autocapitalize="off"
                    autocomplete="off"
                    autocorrect="off"
                    error=""
                    id="disbursementAmount"
                    name="disbursementAmount"
                    spellcheck="false"
                    type="text"
                    value="23.45"
                    warning=""
                  />
                </div>
                <div
                  role="alert"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="paneFooter"
        >
          <div
            class="paneFooterContent"
          >
            <span
              class="renderStart"
              data-test-pane-footer-start="true"
            >
              <button
                class="button default mega"
                data-test-cancel-button="true"
                type="button"
              >
                <span
                  class="inner"
                >
                  stripes-acq-components.FormFooter.cancel
                </span>
              </button>
            </span>
            <span
              data-test-pane-footer-end="true"
            >
              <button
                class="button primary mega"
                data-test-save-button="true"
                disabled=""
                id="save-voucher-button"
                type="submit"
              >
                <span
                  class="inner"
                >
                  stripes-components.saveAndClose
                </span>
              </button>
            </span>
          </div>
        </div>
      </section>
    </div>
    <div
      class="container"
    >
      <div
        class="cursorHover"
        hidden=""
        style="left: -10px;"
      >
        <div
          class="cursor"
        >
          Â 
        </div>
      </div>
    </div>
  </div>
  <div />
</DocumentFragment>
`;
